web/src/navigate.js

```diff
@@ -46,20 +47,11 @@ export function down(with_centering = false) {
 }
 
 export function to_home() {
-    assert(message_lists.current !== undefined);
-    message_viewport.set_last_movement_direction(-1);
```

Comment:
Why do we delete the `message_viewport.set_last_movement_direction(-1)`? I think that might be important.

Reply:
`last_movement_direction` is only used here:
```js
    if (from_scroll) {
        // If the message you're trying to center on is already in view AND
        // you're already trying to move in the direction of that message,
        // don't try to recenter. This avoids disorienting jumps when the
        // pointer has gotten itself outside the threshold (e.g. by
        // autoscrolling).
        if (is_above && last_movement_direction >= 0) {
            return;
        }
        if (is_below && last_movement_direction <= 0) {
            return;
        }
    }
```

which makes `recenter_view` return early if the message is not visible but the user is moving in the direction of the message. We definitely don't want this.

Reply:
OK.